common/src/main/java/com/wynntils/core/features/overlays/Overlay.java

```diff
@@ -47,6 +65,49 @@ public Overlay(
 
     public abstract void render(PoseStack poseStack, float partialTicks, Window window);
 
+    @Override
+    public final void updateConfigOption(ConfigHolder configHolder) {
+        // if user toggle was changed, enable/disable feature accordingly
+        if (configHolder.getFieldName().equals("userEnabled")) {
+            if (isUserEnabled()) {
+                OverlayManager.enableOverlays(List.of(this), true);
+            } else {
+                OverlayManager.disableOverlays(List.of(this));
+            }
+            return;
+        }
+
+        // otherwise, trigger regular config update
+        onConfigUpdate(configHolder);
+    }
+
+    protected void onConfigUpdate(ConfigHolder configHolder) {}
+
+    /** Registers the overlay's config options. Called by ConfigManager when overlay is loaded */
+    public final void addConfigOptions(List<ConfigHolder> options) {
+        configOptions.addAll(options);
+    }
+
+    /** Gets the name of a feature */
+    @Override
+    public String getTranslatedName() {
+        return getTranslation("name");
+    }
+
+    @Override
+    public String getTranslation(String keySuffix) {
+        return I18n.get("overlay.wynntils." + getNameCamelCase() + "." + keySuffix);
```

Comment:
This makes it sound like overlays do not belong to a feature.

Maybe that is a better solution. Maybe we should offer "features" as well as "overlays", and make both first-class citizens. I think that would be a completely valid choice. I would support a redesign for this end.

But the current design says overlays belong to features. If we want to stick with that, then this should be reflected in the key.